### PR TITLE
libtxc_dxtn{,_s2tc}: remove from nixpkgs + hardware.opengl options

### LINKED ISSUE #82627: nixos/opengl: consider enabling S3TC by default
**Describe the bug**

`hardware.opengl.s3tcSupport` currently defaults to false, with an explanation that S3TC texture compression may require a patent license in some locations.

However, all the information I can find online says that S3TC patents expired in 2017, and a lot of major linux distros now ship S3TC on by default. Apparently this makes a significant difference for gaming.

AFAICT, the original S3TC switch dates back to ~2014, so it was correct at the time. But it looks like we could remove it now, and just turn on S3TC whenever OpenGL is enabled - or at least turn it on by default.

### PR DESCRIPTION
**Pull request by delroth**
###### Motivation for this change

Context: discussion in https://github.com/NixOS/nixpkgs/pull/82630

Mesa has been supporting S3TC natively without requiring these libraries
since the S3TC patent expired in December 2017.

Fixes #82627 .

@GrahamcOfBorg build wine-staging lutris

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/hardware/opengl.nix [hunk 1]
```diff
@@ -12,10 +12,7 @@ let
 
   makePackage = p: pkgs.buildEnv {
     name = "mesa-drivers+txc-${p.mesa.version}";
-    paths =
-      [ p.mesa.drivers
-        (if cfg.s3tcSupport then p.libtxc_dxtn else p.libtxc_dxtn_s2tc)
-      ];
+    paths = [ p.mesa.drivers ];
```
Comment (vcunat) [inline nixos/modules/hardware/opengl.nix]: I see no need `buildEnv` from a single package.  Perhaps `makePackage = p: p.mesa.drivers` or even scrap the function completely.

Comment (delroth) [reply nixos/modules/hardware/opengl.nix]: Done.

**Comment by Ekleog** [comment]
@delroth I wanted to review and hopefully merge this, but it looks like merge conflicts arose… could you rebase? :)

**Comment by delroth** [comment]
Rebased.

**Comment by delroth** [comment]
@GrahamcOfBorg build wine-staging lutris

**Comment by Ekleog** [comment]
Looks good to me, thank you!

**Pull request merged**